### PR TITLE
Added JsonSerializerOptions parameter to Set and SetAsync methods.

### LINKED ISSUE #222: Have more JSON serialization controls on JsonCommands Set methods.
**NRedisStack Version**: 0.10.1

**Redis Stack Version**: 7.2.3

**Description**: 
We may need more JSON serialization controls on how object is serialized. It should be easy to add a `JsonSerializerOptions` parameter to `Set`, `SetAsync`, `Merge` and `MergeAsync` method of `JsonCommands` type.

### PR DESCRIPTION
**Pull request by RickyLin**
Closed #222

### CODE REVIEW

File: tests/NRedisStack.Tests/Person.cs [hunk 1]
```diff
@@ -4,6 +4,6 @@ public class Person
     {
         public string? Name { get; set; }
         public int Age { get; set; }
-        public DateTime? Birthday;
+        public DateTime? Birthday { get; set; }
```
Comment (RickyLin) [inline tests/NRedisStack.Tests/Person.cs:7]: Hi @shacharPash ,
I made Birthday a field other than a property on purpose, so it won't be serialized and deserialized by default. That's the reason the JsonSerializerOptions parameter is useful in Set and Merge method if developers want more control on serialization, in the test case, want the Birthday field to be serialized.

**Comment by shacharPash** [comment]
@RickyLin Thanks for your contributing!
I see that there are some failed tests, do you want to fix it?

**Comment by RickyLin** [comment]
Hi @shacharPash,
Those tests were passed after you reverted the `Birthday` field. I can add a comment on that field to explain why it's a field other than a propery if you think necessary.

**Comment by RickyLin** [comment]
Hi @shacharPash,
I'm not very familar with the process, I saw this "All checks have passed" message in this PR, did you mean there are other failed tests besides tests under this "All checks have passed"?
image

**Review by shacharPash — APPROVED** [review]
LGTM 👍

**Pull request merged**